pages/index.tsx

```diff
@@ -0,0 +1,584 @@
+      const apiKeyTypeLinkId = await deep.id("@deep-foundation/openai", "ApiKey");
+      const conversationTypeLinkId = await deep.id("@deep-foundation/chatgpt", "Conversation");
+      const systemTypeLinkId = await deep.id("@deep-foundation/chatgpt", "System");
+      const authorTypeLinkId = await deep.id('@deep-foundation/messaging', 'Author');
+      const messagingTreeId = await deep.id('@deep-foundation/messaging', 'MessagingTree');
+
+      console.log("before insert sound link", sounds);
+      console.log("flakeed2");
+      const { data: [{ id: soundLinkId }] } = await deep.insert(sounds.map((sound) => ({
+        type_id: recordTypeLinkId,
+        in: {
+          data: [{
+            type_id: containTypeLinkId,
+            from_id: audioRecordsLinkId,
+          }]
+        },
+        out: {
+          data: [
+            {
+              type_id: containTypeLinkId,
+              to: {
+                data: {
+                  type_id: soundTypeLinkId,
+                  string: { data: { value: sound.record["recordDataBase64"] } },
+                  out: {
+                    data: [
+                      {
+                        type_id: containTypeLinkId,
+                        to: {
+                          data: {
+                            type_id: mimetypeTypeLinkId,
+                            string: { data: { value: sound.record["mimeType"] } },
+                          }
+                        }
+                      },
+                      {
+                        type_id: containTypeLinkId,
+                        to: {
+                          data: {
+                            type_id: formatTypeLinkId,
+                            string: { data: { value: sound.record["mimeType"] === "audio/webm;codecs=opus" ? "webm" : "aac" } },
+                          }
+                        }
+                      }
+                    ]
+                  }
+                }
+              }
+            },
+            {
+              type_id: containTypeLinkId,
+              to: {
+                data: {
+                  type_id: durationTypeLinkId,
+                  number: { data: { value: sound.record["msDuration"] } },
+                }
+              }
+            },
+            {
+              type_id: containTypeLinkId,
+              to: {
+                data: {
+                  type_id: startTimeTypeLinkId,
+                  string: { data: { value: sound.startTime } },
+                }
+              }
+            },
+            {
+              type_id: containTypeLinkId,
+              to: {
+                data: {
+                  type_id: endTimeTypeLinkId,
+                  string: { data: { value: sound.endTime } },
+                }
+              }
+            }]
+        }
+      })));
+      console.log("flakeed1");
+      console.log("soundLinkId", soundLinkId)
+
+      const { data: [isGcloudAuthKeyLinkId] } = await deep.select({
```

Comment:
Why do you looking for it every time if it is passed at the start and you can store  gcloudAuthKeyLinkId in the local store?